core/parser.mly

```diff
@@ -625,113 +595,25 @@ perhaps_exps:
-| unary_expression INFIX9 unary_expression
-| unary_expression INFIXR9 infixr_9                            { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixl_9:
-| infixr_9                                                     { $1 }
-| infixl_9 INFIXL9 infixr_9                                    { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixr_8:
-| infixl_9                                                     { $1 }
-| infixl_9 INFIX8  infixl_9
-| infixl_9 INFIXR8 infixr_8                                    { infix_appl  ~ppos:$loc $1 $2            $3 }
-| infixl_9 COLONCOLON infixr_8                                 { infix_appl' ~ppos:$loc $1 BinaryOp.Cons $3 }
-
-infixl_8:
-| infixr_8                                                     { $1 }
-| infixl_8 INFIXL8 infixr_8                                    { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixr_7:
-| infixl_8                                                     { $1 }
-| infixl_8 INFIX7  infixl_8
-| infixl_8 INFIXR7 infixr_7                                    { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixl_7:
-| infixr_7                                                     { $1 }
-| infixl_7 INFIXL7 infixr_7                                    { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixr_6:
-| infixl_7                                                     { $1 }
-| infixl_7 INFIX6  infixl_7
-| infixl_7 INFIXR6 infixr_6                                    { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixl_6:
-| infixr_6                                                     { $1 }
-| infixl_6 INFIXL6 infixr_6                                    { infix_appl  ~ppos:$loc $1 $2                  $3 }
-| infixl_6 MINUS infixr_6                                      { infix_appl' ~ppos:$loc $1 BinaryOp.Minus      $3 }
-| infixl_6 MINUSDOT infixr_6                                   { infix_appl' ~ppos:$loc $1 BinaryOp.FloatMinus $3 }
-/* HACK: the type variables should get inserted later... */
-| infixl_6 BANG infixr_6                                       { infix_appl  ~ppos:$loc $1 "!" $3         }
-
-infixr_5:
-| infixl_6                                                     { $1 }
-| infixl_6 INFIX5  infixl_6
-| infixl_6 INFIXR5 infixr_5                                    { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixl_5:
-| infixr_5                                                     { $1 }
-| infixl_5 INFIXL5 infixr_5                                    { infix_appl ~ppos:$loc $1 $2 $3 }
-
-infixr_4:
-| infixl_5                                                     { $1 }
-| infixl_5 INFIX4    infixl_5
-| infixl_5 INFIXR4   infixr_4                                  { infix_appl ~ppos:$loc $1 $2 $3 }
-| infixr_5 EQUALSTILDE regex                                   { let r, flags = $3 in
+| unary_expression OPERATOR                                    { unary_appl ~ppos:$loc (UnaryOp.Name $2) $1 } /* TODO(dhil): This is more general than the previous rule as in it will permit more expressions to be chained together. */
```

Comment:
I wonder whether it's worth getting rid of the comment here and opening an issue, if it's problematic?

Reply:
I have thought some more about it. It shouldn't be a problem. I think I eventually convinced myself that it allows no longer sentences than before to be parsed. What I would like to do is to collapse the whole expression grammar into a single `expression:` rule. It would require a bit of experimentation which I will defer for later.